src/include/miopen/interpolate/problem_description.hpp

```diff
@@ -0,0 +1,256 @@
+                       const miopenInterpolateMode_t mode_,
+                       const bool align_corners_,
+                       bool is_fwd_)
+        : scaleFactorsDesc(scaleFactorsDesc_),
+          mode(mode_),
+          align_corners(align_corners_),
+          is_fwd(is_fwd_)
+    {
+        IsValidMode();
+    }
+
+    const TensorDescriptor& GetScaleFactorsDesc() const { return scaleFactorsDesc; }
+    miopenInterpolateMode_t GetMode() const { return mode; }
+    bool GetAlignCorners() const { return align_corners; }
+
+    bool IsValidMode() const
+    {
+        if(mode != MIOPEN_INTERPOLATE_MODE_NEAREST && mode != MIOPEN_INTERPOLATE_MODE_LINEAR &&
+           mode != MIOPEN_INTERPOLATE_MODE_BILINEAR && mode != MIOPEN_INTERPOLATE_MODE_TRILINEAR &&
+           mode != MIOPEN_INTERPOLATE_MODE_BICUBIC)
+        {
+            std::cout << "MODE: " << mode << std::endl;
+            MIOPEN_THROW(miopenStatusBadParm, "Interpolate: Invalid mode.");
+        }
+        return true;
+    }
+
+protected:
+    TensorDescriptor scaleFactorsDesc;
+    miopenInterpolateMode_t mode;
+    bool align_corners = false;
+    bool is_fwd;
+
+    NetworkConfig MakeForwardNetworkConfig() const;
+};
+
+struct FwdProblemDescription : ProblemDescription
+{
+    FwdProblemDescription(const TensorDescriptor& inputDesc_,
+                          const TensorDescriptor& outputDesc_,
+                          const TensorDescriptor& scaleFactorsDesc_,
+                          const miopenInterpolateMode_t mode_,
+                          const bool align_corners_)
+        : ProblemDescription(scaleFactorsDesc_, mode_, align_corners_, true)
+    {
+        inputDesc  = inputDesc_;
+        outputDesc = outputDesc_;
+        IsValidDims();
+        IsValidLength();
+    }
+
+    const TensorDescriptor& GetInputDesc() const { return inputDesc; }
+    const TensorDescriptor& GetOutputDesc() const { return outputDesc; }
+
+    bool IsValidLength() const
+    {
+        if(inputDesc.GetSize() < 3 || inputDesc.GetSize() > 5)
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "Interpolate: Input tensor size < 3 or > 5 is not valid.");
+        }
+
+        if(outputDesc.GetSize() < 3 || outputDesc.GetSize() > 5)
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "Interpolate: Output tensor size < 3 or > 5 is not valid.");
+        }
+
+        if((outputDesc.GetSize() - 2) != scaleFactorsDesc.GetElementSize())
+        {
+            if(mode != MIOPEN_INTERPOLATE_MODE_NEAREST)
+            {
+                MIOPEN_THROW(
+                    miopenStatusBadParm,
+                    "Interpolate: Output tensor size and scale factors length do not match.");
+            }
+        }
+        return true;
+    }
+
+    bool IsValidDims() const
+    {
+        if(mode == MIOPEN_INTERPOLATE_MODE_LINEAR)
+        {
+            if(inputDesc.GetSize() != 3 || outputDesc.GetSize() != 3)
```

Comment:
can remove `outputDesc.GetSize() != 3` check if you checked inputDesc.GetSize() == outputDesc.GetSize() like above comment.